src/pages/analytics/Analytics.js

```diff
@@ -66,21 +66,16 @@ class Analytics extends Page {
         this.requestTaskSummary()
     }
 
-    componentWillReceiveProps(nextProps) {
+    UNSAFE_componentWillReceiveProps(nextProps) {
         const nextState = {}
 
         Object.keys(nextProps).forEach(property => {
-            if (
-                nextProps.hasOwnProperty(property) &&
-                Analytics.STATE_PROPERTIES.includes(property)
-            ) {
+            if (Analytics.STATE_PROPERTIES.includes(property)) {
                 nextState[property] = nextProps[property]
             }
         })
 
-        if (nextState !== {}) {
-            this.setState(nextState)
-        }
+        this.setState(nextState)
```

Comment:
Why did you remove the condition?
If the condition is always true, I think we could omit this whole method (why should we worry about states that the component doesn't care about anyway) because every update will always include props that are part of this component

Reply:
Applies to all page components

Reply:
After looking at how the components are rendered, I think it's safe to remove this method from all page components.